Migration fixes

niwinz opened the pull request.

niwinz committed:
:bug: Add migrations to fix wrongly migrated data

Also port the migration introduced in main branch
for the recent hotfix

niwinz committed:
:bug: Fix wrong type hints

superalex commented on common/src/app/common/pages/migrations.cljc:
```diff
@@ -291,83 +282,91 @@
+              (and (cph/image-shape? object)
+                   (contains? object :fills)
+                   (= 1 (count (:fills object)))
+                   (= (:fill-color object) (get-in object [:fills 0 :fill-color])))
```
I would also check the fill-color and the opacity are the default values (1 and #B1B2B5)

niwinz replied:
the main idea here is checking if the current :fills has the same value as prev and no more fills are added. I think we don't need more checks here, because previously we only have a single hidden fill-color (and fill-opacity) on images, and the previous migration creates the :fills with single element, so checking the color matches and has a single element is enough

niwinz committed:
:paperclip: Set explicit clojure version on frontend and backend

niwinz committed:
:paperclip: Add analyze-file helper to srepl.main namespace

niwinz committed:
:sparkles: Improve migrate-data function (file data migrations)

This will enable the ability to apply some migration to a specific
file from the Server REPL.

niwinz commented on backend/src/app/srepl/main.clj:
```diff
@@ -66,86 +67,48 @@
-                verbose? false
-                stop-on-error? true
-                max-chunks ##Inf}}]
+;; (defn check-image-shapes
```
This function is commented just as example, I prefer have this code commented here instead of removing it.

hirunatan replied:
this may be explained in a comment, to avoid someone deleting this in the future

niwinz replied:
No one is going to touch this file unless it is needed, and when it needed. I mean, this is the namespace where we leave some functions that need to be executed on the server REPL (scripts). This is not so important to leave comments right now I think, I just saying that I don't want to remove it right now, in future it can be removed without any problem.
